core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1386,6 +1386,12 @@ abstract class RDD[T: ClassTag](
   }
 
   /**
+   * @return false if and only if the RDD contains no elements at all. Note that an RDD
+   *         may be empty even when it has at least 1 partition.
+   */
+  def nonEmpty(): Boolean = !isEmpty()
```

Comment:
This isn't worth adding. I know Scala has both.
An API change would certainly need a JIRA, but this should be closed.